### ISSUE TITLE
Make mapping more flexible/configurable

### ISSUE THREAD

**Issue by psren** [EventSaucePHP/DoctrineMessageRepository#2]
I'd like to add to add 2 more fields to my messages-table.

- Version (e.g. `0`)
- AggregateName (e.g. `document`)

## Why Version?

I want to guard from concurrency issues by setting a PK on `(aggregate_root_id, version)`

## Why AggregateName?

I want a single message stream (one table with all aggregates) and want to maintain the map (`AggregateName` => `Class`) in my application.

## What now

Should I implement my own repository?
Or provide a PR?

**Comment by frankdejonge** [comment]
You can add a new class to this repo with a pr, maybe DoctrineMessageRepositoryWithVersionGuard?

**Comment by frankdejonge** [comment]
This repository is more meant to be a reference implementation, people can implement something like this pretty easy and suit all of their needs. I prefer that from having one implementation that tries to do everything.

**Comment by psren** [comment]
> You can add a new class to this repo with a pr, maybe DoctrineMessageRepositoryWithVersionGuard?

Mhh, after sleeping a night over it, I think it's so easy to implement it.
Maybe should not be part of the repository.

If you want it in here, I'm happy to provide it. If not, you can just close this issue :-)

Adding it has some problems. I could either:

- Duplicate Code (which has to be maintained)
- Abstract the Mapping (which is overkill?)

**Comment by frankdejonge** [comment]
Yeah, I agree. This interface is so small and making it generic makes it needlessly complex.
